src/main/kotlin/org/jlleitschuh/gradle/ktlint/Reporter.kt

```diff
@@ -0,0 +1,44 @@
+// Collection of functions that applies output Ktlint reporter
+
+package org.jlleitschuh.gradle.ktlint
+
+import org.gradle.api.Project
+import org.gradle.api.tasks.JavaExec
+import org.gradle.util.GFileUtils
+import java.io.File
+
+/**
+ * Supported reporter types.
+ */
+enum class ReporterType(val reporterName: String, val availableSinceVersion: String, val fileExtension: String) {
+    PLAIN("plain", "0.9.0", "txt"),
+    PLAIN_GROUP_BY_FILE("plain?group_by_file", "0.9.0", "txt"),
+    CHECKSTYLE("checkstyle", "0.9.0", "xml"),
+    JSON("json", "0.9.0", "json");
+}
+
+/**
+ * Apply reporter to the task.
+ */
+fun JavaExec.applyReporter(target: Project, extension: KtlintExtension) {
+    if (isReportAvailable(extension.version, extension.reporter.availableSinceVersion)) {
+        this.args("--reporter=${extension.reporter.reporterName}")
+        this.standardOutput = createReportOutputDir(target, extension).outputStream()
+    } else {
+        target.logger.info("Reporter is not available in this ktlint version")
```

Comment:
I've also initially thought about debug level, but gradle, on debug level, produces too much output. This message can be missed. And this message, IMHO, is more informational, then debug.